test/unit/org/apache/cassandra/service/ActiveRepairServiceTest.java

```diff
@@ -387,10 +407,13 @@ public void testRejectWhenPoolFullStrategy() throws InterruptedException
             {
                 // expected
             }
+
             // allow executing tests to complete
             blocked.signalAll();
-            completed.await(10, TimeUnit.SECONDS);
+            completed.await(11, TimeUnit.SECONDS);
+
             // Submission is unblocked
+            Thread.sleep(100);
```

Comment:
Why do we need this second sleep?

Reply:
Again the same error is produced if we don't wait. My guess being between task completion and TP readiness there is a small window the queue/TP interaction is broken.